sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonIUD.java

```diff
@@ -0,0 +1,376 @@
+      if (columnToValueMapping.containsKey(column)) {
+        columnToValueMapping.get(column).add(value);
+      } else {
+        Set<String> columnValues = new HashSet<>();
+        columnValues.add(value);
+        columnToValueMapping.put(column, columnValues);
+      }
+    } else {
+      Map<String, Set<String>> columnToValueMapping = new HashMap<>();
+      Set<String> columnValues = new HashSet<>();
+      columnValues.add(value);
+      columnToValueMapping.put(column, columnValues);
+      filterColumnToValueMapping.put(path, columnToValueMapping);
+    }
+  }
+
+  /**
+   * This method will convert the given columnToValue mapping into expression object
+   * If columnToValue mapping have following entries:
+   * name --> {karan, kunal, vikram}
+   * age --> {24}
+   * the expression will look like this for above entries:
+   * ((name = karan || name = kunal || name = vikram) && (age = 24))
+   */
+  private Expression getExpression(String path, Map<String, Set<String>> columnToValueMapping)
+      throws IOException {
+    File[] indexFiles = getCarbonIndexFile(path);
+    Schema schema = CarbonSchemaReader.readSchema(indexFiles[0].getAbsolutePath()).asOriginOrder();
+    Field[] fields = schema.getFields();
+    List<Expression> listOfExpressions = new ArrayList<>();
+    for (Map.Entry<String, Set<String>> column : columnToValueMapping.entrySet()) {
+      DataType dataType = getColumnDataType(fields, column.getKey());
+      List<Expression> listOfOrExpressions = new ArrayList<>();
+      for (String value : column.getValue()) {
+        listOfOrExpressions.add(
+            new EqualToExpression(new ColumnExpression(column.getKey(), dataType),
+                new LiteralExpression(value, dataType)));
+      }
+      Expression OrFilterExpression = null;
+      if (listOfOrExpressions.size() > 0) {
+        OrFilterExpression = listOfOrExpressions.get(0);
+      }
+      for (int i = 1; i < listOfOrExpressions.size(); i++) {
+        OrFilterExpression = new OrExpression(OrFilterExpression, listOfOrExpressions.get(i));
+      }
+      listOfExpressions.add(OrFilterExpression);
+    }
+    Expression filterExpression = null;
+    if (listOfExpressions.size() > 0) {
+      filterExpression = listOfExpressions.get(0);
+    }
+    for (int i = 1; i < listOfExpressions.size(); i++) {
+      filterExpression = new AndExpression(filterExpression, listOfExpressions.get(i));
+    }
+    return filterExpression;
+  }
+
+  private int getColumnIndex(Field[] fields, String column) {
+    int index = -1;
+    for (Field field : fields) {
+      if (field.getFieldName().equals(column)) {
+        index = field.getSchemaOrdinal();
+        break;
+      }
+    }
+    if (index == -1) {
+      throw new RuntimeException("ColumnName doesn't exists");
+    }
+    return index;
+  }
+
+  private File[] getCarbonIndexFile(String path) {
+    File[] indexFiles = new File(path).listFiles(new FilenameFilter() {
```

Comment:
make this a  recursive listing so that it can work for transactional tables also

Reply:
Done